tinygrad/mlops.py

```diff
@@ -148,7 +149,7 @@ def backward(self, grad_output:LazyBuffer) -> Tuple[None, Optional[LazyBuffer],
 class Sum(Function):
   def forward(self, x:LazyBuffer, new_shape:Tuple[int, ...]) -> LazyBuffer:
     self.input_shape = x.shape
-    return x.r(ReduceOps.SUM, new_shape)
+    return x.cast(least_upper_dtype(x.dtype, dtypes.uint8)).r(ReduceOps.SUM, new_shape) # no reduce on bools TODO should be int8? int32 is safer? TODO should this be fixed upstream in sparse_cat_cross_entropy?
```

Comment:
```
>>> jnp.array([True, False, False], dtype="int8").sum()
Array(1, dtype=int32)
>>> torch.tensor([True, False, False], dtype=torch.int8).sum().dtype
torch.int64
```
hmm, what's the behavior you want?

Reply:
if there's a bool sum you probably want an implicit cast to a numeric?
otherwise these things happen https://gist.github.com/Qazalin/f71826bbc685717aed08ec9fd4a8410c

Reply:
i understand we want to upcast for boolean sum, i meant to ask what's your thought on which dtype summation resulted in which dtype, and how did you reach that decision

Reply:
imo int32 is the best practical choice but theoretically just relying on uint8 will _work_ but it might too small